### ISSUE TITLE
Error when edit a container

### ISSUE THREAD

**Issue by fmontes** [dotCMS/core#23335]
### To Reproduce
1. Go to edit a container with content types
2. Check the browser console
3. See error

### Expected behavior
It should not show any errors.

### Screenshots
<img width="824" alt="image" src="https://user-images.githubusercontent.com/751424/200963129-c2485834-6038-49c0-bce9-5212440ed8f5.png">

### Acceptance criteria

- [ ] No error in the console

**Comment by fmontes** [comment]
The architecture of these components have multiple state management issues which is leading to unexpecting results and errors.

I'm studying the code to come up with a plan on improve this.

**Comment by zulqarnainvd** [comment]
@fmontes I discussed this issue with Motassim, who said the edit container work is not completed.

**Comment by fmontes** [comment]
Yes, is not complete, but also we are facing some errors I think because of the arq that I'm researching on how to solve them.

**Comment by fmontes** [comment]
After doing some research and look at the code, here are the problems that I identify:

### FormArray 
We can't pass `FormArray` to controls, [here is more info about it](https://github.com/angular/angular/issues/31958#issuecomment-529220939).

So, my recommendation here is basically what the comment said in GitHub, just create a normal component that get the data, and we can add it to the `FormArray` using `push` or `insert` method.

So in short, [this component](https://github.com/dotCMS/core/blob/ba4b429ee68847821088864b24c461f707943859/core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts) should not implement `ControlValueAccessor` just receive and emit.

### Typed Forms
Angular 14 implements types forms, which can help us to catch type errors here, we should use it.

**Comment by zulqarnainvd** [comment]
@fmontes I checked your example but in our case, there is a child component where I need to pass the form because we need to split the code component. would you please check the example I created and let me know if it works for you?

https://stackblitz.com/edit/form-array-in-child-zulqarnain?file=src/app/hello.component.html

**Comment by fmontes** [comment]
Yes, your example is the same as what I did here:

<img width="683" alt="image" src="https://user-images.githubusercontent.com/751424/203065705-09388ad7-3be0-42a5-bbe1-0d1ddcabc979.png">

Sorry I shared the wrong link, here is the link to the code:

https://stackblitz.com/edit/component-store-reactive-forms?file=src/app/form/form.component.html
